crates/accelerate/src/sabre/route.rs

```diff
@@ -383,13 +418,33 @@ impl<'a, 'b> RoutingState<'a, 'b> {
             }
         }
 
+        if let Some(DepthHeuristic { .. }) = self.heuristic.depth {
+            let curr_depth = self.circuit_depth();
+
```

Comment:
This is a really cool development! I have two minor questions:

* Would it be possible to update the current maximum depth as gates are routed; possibly this would be a bit faster than computing the maximum value each time?

* The current computation prioritizes SWAPs that do not increase the current maximum depth to those that do. In the case that multiple swaps-without-depth-increase are available, would it make sense to further prioritize SWAPs that connect a pair of qubits with lowest depths?

Reply:
Thank you for the feedback—I'm glad you found the development interesting!

1. Yes, updating the current maximum depth as gates are routed is definitely possible and a great idea. It would be faster than computing the maximum value each time. I'll implement this improvement—thanks for the suggestion!

2. It makes sense to prioritize further SWAPs that connect a pair of qubits with the lowest depths, and often multiple swaps without depth increase are available. My current method addresses this is to calculating the "true depth" after inserting a SWAP. This involves determining the immediate gates that can be executed following the swap and then computing the new depth based on those gates. I’ve implemented this in my Python version, where I executed the experiments, and am currently working on integrating it into the Rust side.